### PR TITLE
RIFEX-60 Register a new domain

### PR DESCRIPTION
**Pull request by JONAF2103**
This PR introduce the ui for registration, this PR affects only ui and adds some extra service methods. 
We have some screens where the user can search for a domain to register, create a commitment and confirm the domain registration, also can see the registered domain in a list. All that allow the user to own a free domain by making transactions. We have a demo here about how the process works.

test

### CODE REVIEW

File: ui/app/rif/constants/index.js [hunk 1]
```diff
@@ -36,4 +36,9 @@ export const luminoNodeIconProps = {
 export const rifStorageIconProps = {
     color: '#AD3232',
     icon: faArchive,
-}
\ No newline at end of file
+}
+
+export const registrationTimeouts = {
```
Comment (marcosmartinez7) [inline ui/app/rif/constants/index.js:41]: Can you please put some comments to reflect what those numbers means?

File: ui/app/rif/pages/domains/register/index.js [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+
+  showConfirmTransactionPage (callback, payload) {
+    this.props.getUnapprovedTransactions()
+      .then(latestTransaction => {
+        this.props.showTransactionConfirmPage({
+          id: latestTransaction.id,
+          unapprovedTransactions: latestTransaction,
+          afterApproval: {
+            action: (payload) => {
+              if (callback) {
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/domains/register/index.js]: Can we avoid this payload check? If we execute callback(payload) and payload is undefined or falsy, then the callback function should handle it exactly the same as without arguments, right?

**Review by marcosmartinez7 — APPROVED** [review]

**Comment by marcosmartinez7** [comment]
In general: great work :)

Just added some comments that improves the code i think. 

Also, it would be nice to add more human readable messages during registration. Think in a normal user that doesnt know about how RNS works. Since we have multiple registration phases and and the user must wait some time, maybe we can improve the app feedback.

File: ui/app/rif/pages/domains/register/index.js [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+            className="checkmark"
+            xmlns="http://www.w3.org/2000/svg"
+            width="96"
+            height="96"
+            viewBox="0 0 52 52">
+            <circle className="checkmark__circle" cx="26" cy="26" r="25" fill="none" />
+            <path className="checkmark__check" fill="none" d="M14.1 27.2l7.1 7.2 16.7-16.8" />
+          </svg>
+          <div>
+            We are ready!!!
```
Comment (marcosmartinez7) [inline ui/app/rif/pages/domains/register/index.js]: Lets put just one exclamation mark :P

File: ui/app/rif/constants/index.js (hunk 1 above)
Comment (JONAF2103) [reply ui/app/rif/constants/index.js:41]: Sure

File: ui/app/rif/pages/domains/register/index.js (hunk 2 above)
Comment (JONAF2103) [reply ui/app/rif/pages/domains/register/index.js]: Yeah i think you are right.

File: ui/app/rif/pages/domains/register/index.js (hunk 3 above)
Comment (JONAF2103) [reply ui/app/rif/pages/domains/register/index.js]: All this together with the other feedback issues are related to the designs, i just made this pretty similar to the designs, actually i did it as most similar as possible, so maybe that's the issue, i think they are going to put this through a designer so after that these kind of issues are going to disapear

**Pull request merged**